File: ItemNest.Domain/Entities/ContactRequest.cs
using ItemNest.Domain.Enums;

namespace ItemNest.Domain.Entities;

public class ContactRequest
{
    public Guid Id { get; set; }

    public Guid RequesterUserId { get; set; }
    public AppUser RequesterUser { get; set; } = null!;

    public Guid PostOwnerUserId { get; set; }
    public AppUser PostOwnerUser { get; set; } = null!;

    public Guid ItemPostId { get; set; }
    public ItemPost ItemPost { get; set; } = null!;

    public string Message { get; set; } = string.Empty;
    public ContactRequestStatus Status { get; set; } = ContactRequestStatus.Pending;

    public DateTime CreatedAt { get; set; }
    public DateTime? RespondedAt { get; set; }
}

File: ItemNest.Api/itemnest-front/src/types/contactRequest.ts
export interface CreateContactRequestRequest {
  itemPostId: string;
  message: string;
}

export interface ContactRequestItem {
  id: string;
  itemPostId: string;
  itemPostTitle: string;
  requesterUserId: string;
  requesterFullName: string;
  requesterEmail: string | null;
  postOwnerUserId: string;
  postOwnerFullName: string;
  postOwnerEmail: string | null;
  message: string;
  status: number;
  createdAt: string;
  respondedAt: string | null;
}